File: modules/lib/lib-project/src/test/resources/test/GetProjectHandlerTest.js
var assert = require('/lib/xp/testing.js');
var project = require('/lib/xp/project.js');

var getProjectExpected = {
    id: 'myproject',
    displayName: 'project display name',
    description: 'project description',
    parents: [],
    permissions: {
        owner: [
            'user:system:owner2',
            'user:system:owner1'
        ],
        viewer: [
            'user:system:viewer1'
        ]
    },
    readAccess: {
        'public': true
    }
};

exports.getProject = function () {
    createProject({owner: ['user:system:owner2', 'user:system:owner1'], viewer: ['user:system:viewer1']});

    var result = project.get({
        id: 'myproject'
    });

    assert.assertJsonEquals(getProjectExpected, result);
};

var getProjectWithoutPermissionsExpected = {
    id: 'myproject',
    displayName: 'project display name',
    description: 'project description',
    parents: [],
    permissions: {},
    readAccess: {
        'public': true
    }
};

exports.getProjectWithoutPermissions = function () {
    createProject(null);

    var result = project.get({
        id: 'myproject'
    });

    assert.assertJsonEquals(getProjectWithoutPermissionsExpected, result);
};

exports.getProjectNotExist = function () {
    createProject();

    var result = project.get({
        id: 'myproject1'
    });

    assert.assertJsonEquals(null, result);
};

exports.getProjectNull = function () {
    createProject();

    try {
        project.get({
            id: null
        });

        throw new Error('IllegalArgumentException should be thrown.');

    } catch (e) {
        assert.assertEquals('Project name is required', e.getMessage());
    }
};


function createProject(permissions) {
    project.create({
        id: 'myproject',
        displayName: 'project display name',
        description: 'project description',
        readAccess: {public: true},
        permissions: permissions
    });
}

File: modules/lib/lib-project/src/test/java/com/enonic/xp/lib/project/GetProjectHandlerTest.java
package com.enonic.xp.lib.project;

import org.junit.jupiter.api.Test;

class GetProjectHandlerTest
    extends BaseProjectHandlerTest
{
    @Test
    void getProject()
    {
        runFunction( "/test/GetProjectHandlerTest.js", "getProject" );
    }

    @Test
    void getProjectWithoutPermissions()
    {
        runFunction( "/test/GetProjectHandlerTest.js", "getProjectWithoutPermissions" );
    }

    @Test
    void getProjectNull()
    {
        runFunction( "/test/GetProjectHandlerTest.js", "getProjectNull" );
    }

    @Test
    void getProjectNotExist()
    {
        runFunction( "/test/GetProjectHandlerTest.js", "getProjectNotExist" );
    }

}
